protocol/lib/metrics/lib.go

```diff
@@ -90,3 +93,41 @@ func ModuleMeasureSinceWithLabels(
 		),
 	)
 }
+
+func isAllowedExecutionMode(
+	ctx sdk.Context,
+	allowedModes []sdk.ExecMode,
+) bool {
+	contextExecMode := ctx.ExecMode()
+	for _, mode := range allowedModes {
+		if contextExecMode == mode {
+			return true
+		}
+	}
+	return false
+}
+
+func EmitTelemetryWithLabelsForExecMode(
+	ctx sdk.Context,
+	allowedModes []sdk.ExecMode,
+	telemtryFuncWithLabels TelemetryEmitWithLabelsFunc,
+	key []string,
+	val float32,
+	labels []gometrics.Label,
+) {
+	if isAllowedExecutionMode(ctx, allowedModes) {
+		telemtryFuncWithLabels(key, val, labels)
+	}
+}
+
+func EmitTelemetryForExecMode(
```

Comment:
Technically no need to do this since you can ...kwargs the labels parameter and then always use the `metrics.SetXXXXWithLabels` fn